src/confcom/azext_confcom/tests/latest/test_confcom_tar.py

```diff
@@ -169,27 +184,15 @@ def test_arm_template_with_parameter_file_clean_room_tar(self):
             custom_arm_json_default_value, ""
         )[0]
 
-        # save the tar file for the image in the testing directory
-        client = docker.from_env()
-        image = client.images.get("nginx:1.22")
-        tar_mapping_file = {"nginx:1.22": self.image_path}
-        # Note: Class setup and teardown shouldn't have side effects, and reading from the tar file fails when all the tests are running in parallel, so we want to save and delete this tar file as a part of the test. Not as a part of the testing class.
-        f = open(self.image_path, "wb")
-        for chunk in image.save(named=True):
-            f.write(chunk)
-        f.close()
-        client.close()
         tar_mapping_file = {"nginx:1.22": self.image_path}
         try:
             clean_room_image.populate_policy_content_for_all_images(
                 tar_mapping=tar_mapping_file
             )
-        except:
+            sem.release()
```

Comment:
Also maybe a dumb question, should there be an `acquire()` somewhere before the call the `release()` here?

Reply:
Great points, my understanding for the first question is since the semaphore starts at `4`, we can have up to 4 releases without acquiring. This usage is just for counting down that all the tests completed before we can delete the tar file. That way only the last test will delete it, all others should run tearDownClass with no effect.

Second question, yes using a `finally` would be better, I'll make that change to this spot and the few others where `sem.release()` is in a try/catch